resolve/index.go

```diff
@@ -104,6 +114,15 @@ func NewRuleIndex(mrslv func(r *rule.Rule, pkgRel string) Resolver) *RuleIndex {
 	}
 }
 
+// NewRuleIndexCrslv creates a new index with CrossResolver capabilities.
+func NewRuleIndexCrslv(mrslv func(r *rule.Rule, pkgRel string) Resolver, crslvs []CrossResolver) *RuleIndex {
```

Comment:
I'm a little worried we'd need to introduce another new constructor if more interfaces like `CrossResolver` are added in the future.

How about instead we add a parameter `exts ...interface{}` to the original constructor, then just pass the full list of languages to that? It can pick out anything that satisfies `CrossResolver` instead of making `fix-update.go` do that.

Technically this is still a breaking change, since the function can't be assigned to a variable of the old type. Calls should still work though, so I think I'm comfortable with that.

Reply:
Unfortunately, `[]language.Language` cannot be passed into `resolve.NewRuleIndex` as an `interface{}` without causing an import cycle. 

Instead, I added `exts ...interace{}` to the original constructor, but `fix-update.go` is still picking out any `lang` that satisfies `CrossResolver`.